### DISCUSSION TITLE
MSBuild Shell Extension

### DISCUSSION THREAD

**Discussion by DevX-Cipher** [dotnet/msbuild#7356] [Show and tell]
hi i made a shell extension for msbuild here is a photo of what it looks like when you right click on .csproj will be adding ..vcxproj support later on for myself i didnt contact you before about this but never got a reply

![Untitled](https://user-images.githubusercontent.com/25472227/151673587-da0ad247-1a2d-4d99-8691-7431e6820976.png)

**Comment by rainersigwald** [comment]
This is super cool, but I think it needs to be a separate project. As I mentioned in https://github.com/dotnet/msbuild/pull/7346#issuecomment-1024318367, Visual Studio's side-by-side installation requirements would make this very complicated to implement as part of the project, and there hasn't been much demand.

But that doesn't mean it can't work really well for some use cases, like yours! So I think it's best as a separate tool that doesn't have to have all of the constraints we operate under in this repo.

You may also want to check out some of the new [Windows 11 Context Menu](https://blogs.windows.com/blog/2021/07/19/extending-the-context-menu-and-share-dialog-in-windows-11/) stuff.

**Reply by DevX-Cipher** [reply]
i was hoping to create something that will generate the .vcxproj file which i can do it with python but not C#

**Reply by Nirmal4G** [reply]
I would like this but MSBuild is a command line tool and only .NET, PHP, C++ are the active participants of MSBuild. That could be why there hasn't been any demand for this. But I can see this going like [TortoiseGit](https://tortoisegit.org) but for MSBuild only if MSBuild expands everywhere.
